Prepare Getting Started with Popup Guide (#2137)
DevExpress/devextreme-documentation: concepts/05 UI Components/Popup/00 Getting Started with Popup/04 Set Visibility.md

> ---

Reviewer: Rename the article to Show and Hide the Popup

##### jQuery

> Use the **visibility** parameter to control whether the popup is hidden or shown by default. With this parameter set to **true**, the popup will displayed by default and cannot be reopened once closed. In order to toggle popup's visibility, you can use [show()](/Documentation/ApiReference/UI_Components/dxPopup/Methods/#show) and [hide()](/Documentation/ApiReference/UI_Components/dxPopup/Methods/#hide) methods. Add a [button](/Documentation/ApiReference/UI_Components/dxButton/) and implement popup toggle in its **onClick** handler. Use the [closeOnOutsideClick](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/#closeOnOutsideClick) do define whether users can close the popup on outside clicks:

Reviewer: - "by default" - not recommended.
- "With this parameter set to **true**, the popup will displayed by default" - this repeats part of the thought from the previous sentence.
- "Add a [button](/Documentation/ApiReference/UI_Components/dxButton/) and implement popup toggle in its **onClick** handler." - this shouldn't be a direct instruction. The show and hide methods can be called in many other ways, so we can only describe how we call them.
- "close the popup on outside clicks" - sounds strange, although I do understand what it says.

```suggestion
To show the Popup, call its [show()](/Documentation/ApiReference/UI_Components/dxPopup/Methods/#show) method. In the code below, this method is called when users click the Open Popup button. You can also enable the [visible](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/#visible) property to show the Popup immediately after creation.

To hide the Popup, call its [hide()](/Documentation/ApiReference/UI_Components/dxPopup/Methods/#hide) method. Users can hide the Popup when they click outside its boundaries. To enable this functionality, set the [closeOnOutsideClick](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/#closeOnOutsideClick) property to **true**.
```

    <!-- tab: index.js -->
    $(function () {
        $("#popup").dxPopup({
            // ...
            visible: true,
            closeOnOutsideClick: true
        });

        const popup = $("#popup").dxPopup("instance");

        $("#button").dxButton({
            text: "Open popup",
            onClick: () => {
                popup.show();
            }
        });
    });

    // ...
 
##### Angular

> Use the **visibile** parameter to control whether the popup is hidden or shown by default. With this parameter set to **true**, the popup will displayed by default and cannot be reopened once closed. In order to toggle popup's visibility two-way bind the **visibile** property to manipulate its value. Add a [button](/Documentation/ApiReference/UI_Components/dxButton/) and implement popup toggle in its **onClick** handler. Use the [closeOnOutsideClick](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/#closeOnOutsideClick) do define whether users can close the popup on outside clicks:

Reviewer: Try to rewrite this paragraph in the manner of the suggestion above.

    <!-- tab: app.component.html -->
    <dx-button
        text="Open popup"
        (onClick)="togglePopup()"

>     >

Reviewer: ```suggestion
        (onClick)="togglePopup()">
```

    </dx-button>
    <dx-popup
        [(visible)]="isPopupVisible"
        [closeOnOutsideClick]="true"

>     >

Reviewer: ```suggestion
        [closeOnOutsideClick]="true">
```

        <div *dxTemplate="let data of 'content'">
            // ...

>         </div>

Reviewer: ```suggestion
        <!-- ... -->
```

    })
    export class AppComponent {
        isPopupVisible: boolean;

        constructor() {
            this.isPopupVisible = true;
        }

        togglePopup(): void {
            this.isPopupVisible = !this.isPopupVisible;
        }
    }

##### Vue

Use the **visibile** parameter to control whether the popup is hidden or shown by default. With this parameter set to **true**, the popup will displayed by default and cannot be reopened once closed. In order to toggle popup's visibility two-way bind the **visibile** property to manipulate its value. Add a [button](/Documentation/ApiReference/UI_Components/dxButton/) and implement popup toggle in its **click** handler. Use the [closeOnOutsideClick](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/#closeOnOutsideClick) do define whether users can close the popup on outside clicks:

    <!-- tab: App.vue -->
    <template>
        <div id="app-container">
            <DxButton
                text="Open popup"
                @click="togglePopup"            
            />
            <DxPopup
                v-model:visible="isPopupVisible"
                :close-on-outside-click="true"            

>             >

Reviewer: ```suggestion
                :close-on-outside-click="true">
```

                <template #content>
                    // ...

>                 </template>            

Reviewer: ```suggestion
                <!-- ... -->
```

            DxButton,
        },
    }
    </script>

##### React

Use the **visibile** parameter to control whether the popup is hidden or shown by default. With this parameter set to **true**, the popup will displayed by default and cannot be reopened once closed. In order to toggle popup's visibility, bind the **visibile** property to a constant and manipulate its value with React **useState** hook. As React does not support two-way data binding, use the [onHiding](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/#onHiding) property to adjust the value of the **visible** property when popup is closed. Add a [button](/Documentation/ApiReference/UI_Components/dxButton/) and implement popup toggle in its **onClick** handler. Use the [closeOnOutsideClick](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/#closeOnOutsideClick) do define whether users can close the popup on outside clicks:

    <!-- tab: App.js -->
    import React, { useState } from 'react';

    import 'devextreme/dist/css/dx.common.css';
    import 'devextreme/dist/css/dx.light.css';
    
    // ...
    import { Button } from 'devextreme-react/button';

    function App() {

>         const [isPopupVisible, setPopupVisivility] = useState(true);

Reviewer: visivility

        const togglePopup = () => {
            setPopupVisivility(!isPopupVisible);
        };
        return (
            <div className="App">
                <Button 
                    text="Open popup"
                    onClick={togglePopup} 
                />
                <Popup
                    visible={isPopupVisible}
                    closeOnOutsideClick={true}
                    onHiding={togglePopup}                

>                 >

Reviewer: ```suggestion
                    onHiding={togglePopup}>
```

>                     // ...

Reviewer: The Popup should use a contentRender, so there shouldn't be omitted nested elements.
